src/transformers/image_processing_utils_fast.py

```diff
@@ -131,3 +121,882 @@ def get_max_height_width(images: List["torch.Tensor"]) -> Tuple[int]:
+        """
+        validate_kwargs(captured_kwargs=kwargs.keys(), valid_processor_keys=self.valid_extra_kwargs)
+
+        do_resize = do_resize if do_resize is not None else self.do_resize
+        size = size if size is not None else self.size
+        default_to_square = kwargs.pop(
+            "default_to_square", self.default_to_square if self.default_to_square is not None else True
+        )
+        size = get_size_dict(size=size, default_to_square=default_to_square) if size is not None else None
+        resample = resample if resample is not None else self.resample
+        do_center_crop = do_center_crop if do_center_crop is not None else self.do_center_crop
+        crop_size = crop_size if crop_size is not None else self.crop_size
+        crop_size = get_size_dict(crop_size, param_name="crop_size") if crop_size is not None else None
+        do_rescale = do_rescale if do_rescale is not None else self.do_rescale
+        rescale_factor = rescale_factor if rescale_factor is not None else self.rescale_factor
+        do_normalize = do_normalize if do_normalize is not None else self.do_normalize
+        image_mean = image_mean if image_mean is not None else self.image_mean
+        image_std = image_std if image_std is not None else self.image_std
+        do_convert_rgb = do_convert_rgb if do_convert_rgb is not None else self.do_convert_rgb
+        return_tensors = "pt" if return_tensors is None else return_tensors
+        device = kwargs.pop("device", None)
+
+        images, image_mean, image_std, size, crop_size, interpolation = self.prepare_process_arguments(
+            images=images,
+            do_resize=do_resize,
+            size=size,
+            resample=resample,
+            do_center_crop=do_center_crop,
+            crop_size=crop_size,
+            do_rescale=do_rescale,
+            rescale_factor=rescale_factor,
+            do_normalize=do_normalize,
+            image_mean=image_mean,
+            image_std=image_std,
+            do_convert_rgb=do_convert_rgb,
+            return_tensors=return_tensors,
+            data_format=data_format,
+            input_data_format=input_data_format,
+            device=device,
+            **kwargs,
+        )
+
+        processed_images = []
+        for image in images:
+            if do_resize:
+                image = self.resize(
+                    image=image,
+                    size=size,
+                    resample=interpolation,
+                )
+
+            if do_center_crop:
+                image = self.center_crop(image, crop_size)
+
+            if do_rescale and do_normalize:
+                # fused rescale and normalize
+                image = self.normalize(image.to(dtype=torch.float32), image_mean, image_std)
+            elif do_rescale:
+                image = image * rescale_factor
+            elif do_normalize:
+                image = self.normalize(image, image_mean, image_std)
```

Comment:
After resizing all images to the same shape we can concatenate them and use the rest of ops in batch, should also give a speedup

Reply:
Yes absolutely! Will add this